### PR TITLE
Implement banned_addons

### PR DESCRIPTION
**Pull request by barncastle**
**Changes proposed:**

-  Implement the ability to ban addons based on Trinity's implementation

**Target branch(es):** Master

**Tests performed:** Built and basic tests done

**Known issues and TODO list:**
-  Hard coded the offset in AddonMgr.cpp whereas Trinity reads from the DBC. This offset doesn't appear to even be required

### CODE REVIEW

**Comment by ShinDarth** [comment]
@barncastle please fix the indentation, set your editor config to be 4 spaces

**Comment by BarbzYHOOL** [comment]
Tested it several times, it works perfectly fine. It can be merged.

PS : @barncastle made this PR after we talked about a dbc and his tool WDBX, and we found out this code was lacking on AC but not on TC. So he kindly ported it to TC for ~~me~~ us all. Thanks a lot.

File: src/game/Server/WorldSession.cpp [hunk 1]
```diff
@@ -1079,9 +1079,18 @@ void WorldSession::SendAddonsInfo()
...
+    {
+	    data << uint32(itr->Id);
+        data.append(itr->NameMD5, sizeof(itr->NameMD5));
+        data.append(itr->VersionMD5, sizeof(itr->VersionMD5));
+        data << uint32(itr->Timestamp);
+        data << uint32(1);  // IsBanned
+    }
 
-    SendPacket(&data);
+SendPacket(&data);
```
Comment (ShinDarth) [inline src/game/Server/WorldSession.cpp]: can you please revert this line change?

File: data/sql/base/db_characters/banned_addons.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  PRIMARY KEY (`Id`),
+  UNIQUE KEY `idx_name_ver` (`Name`, `Version`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
+/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
+/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
+/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;
+
```
Comment (ShinDarth) [inline data/sql/base/db_characters/banned_addons.sql]: we also need an update file for this

**Pull request merged**